src/util/GeoSparqlHelpers.cpp

```diff
@@ -13,64 +15,85 @@ namespace ad_utility {
+  while (*s == ' ') ++s;
+  if (*s++ != '(') return nullptr;
+  while (*s == ' ') ++s;
+  const char* beg = s;
+  for (int i = 1; i <= 2; ++i) {
+    while (*s == ' ') ++s;
+    if (*s == '-') ++s;
+    if (!(isdigit(*s++))) return nullptr;
+    while (isdigit(*s)) ++s;
+    if (*s == '.') {
+      ++s;
+      if (!(isdigit(*s++))) return nullptr;
+      while (isdigit(*s)) ++s;
+    }
+  }
+  while (*s == ' ') ++s;
+  if (*s++ != ')') return nullptr;
+  if (*s++ != '"') return nullptr;
+  return beg;
 }
 
-// Helper function that parses a single WKT Point of the form "POINT(<double>
-// <double>)", where the quote are included, the POINT must be uppercase, there
-// is an arbitrary amount of whitespace after the comma, and arbitrary contents
-// after the final quote. If the string does not match this format, a pair of
-// `quiet_NaN` is returned
+// Helper function that parses a single WKT Point. If the parsing succeeds,
+// returns a pair of latitude and longitude (note that it's the other way round
+// in WKT). Otherwise, return a pair of `quiet_NaN`.
 std::pair<double, double> parseWktPoint(const std::string& p) {
   auto error_result = std::pair(std::numeric_limits<double>::quiet_NaN(),
                                 std::numeric_limits<double>::quiet_NaN());
-  if (!isWktPoint(p)) {
-    return error_result;
-  }
-  const char* beg = p.c_str() + 7;
-  char* end;
-  double lng = std::strtod(beg, &end);
-  if (end == nullptr) return error_result;
-  beg = end + 1;
-  double lat = std::strtod(beg, &end);
-  // Note: When the first condition is false, `end` points to a part of the
-  // string (possibly the terminating 0), so that `*end` is OK. When the second
-  // condition is false, `end` points to `)`, so that end + 1 points to a part
-  // of the string (possibly the terminating 0), so that `*(end + 1)` is OK.
-  if (end == nullptr || *end != ')' || *(end + 1) != '"') return error_result;
+  const char* beg = checkWktPoint(p.data());
+  if (beg == nullptr) return error_result;
+  const char* end = p.data() + p.size();
+  double lng, lat;
+  auto ret = absl::from_chars(beg, end, lng);
+  if (ret.ec == std::errc::invalid_argument) return error_result;
+  beg = ret.ptr + 1;
+  while (*beg == ' ') ++beg;
+  ret = absl::from_chars(beg, end, lat);
+  if (ret.ec == std::errc::invalid_argument) return error_result;
   return std::pair(lat, lng);
 }
 
-// See wktLongitude.
+// Parse longitude from WKT point.
 double wktLongitudeImpl(const std::string& p) {
   auto nan = std::numeric_limits<double>::quiet_NaN();
-  if (!isWktPoint(p)) return nan;
-  const char* beg = p.c_str() + 7;
-  char* end;
-  double lng = std::strtod(beg, &end);
-  return end != nullptr || *(end + 1) == ' ' ? lng : nan;
+  const char* beg = checkWktPoint(p.data());
+  if (beg == nullptr) return nan;
+  const char* end = p.data() + p.size();
+  double lng;
+  auto result = absl::from_chars(beg, end, lng);
+  if (result.ec == std::errc::invalid_argument) return nan;
```

Comment:
This code pattern appears four times, it should be a small function
(`toDoubleOrNan(std::string_view)` for example)

Reply:
Due to the regex parsing, the code is so much simpler now that this problem has gone away.